File: Backend/src/main/java/com/Stage/Dalil_sante/entity/TypeEtablissement.java
package com.Stage.Dalil_sante.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "types_etablissement")
public class TypeEtablissement {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String nom;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private Boolean actif = true;

    public TypeEtablissement() {
    }

    public TypeEtablissement(
            String nom,
            String description,
            Boolean actif
    ) {
        this.nom = nom;
        this.description = description;
        this.actif = actif;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Boolean getActif() {
        return actif;
    }

    public void setActif(Boolean actif) {
        this.actif = actif;
    }
}

File: frontend/src/types/typeEtablissement.ts
/** كيان "TypeEtablissement" حرفيًا كما في entity/TypeEtablissement.java */
export interface TypeEtablissement {
  id: number
  nom: string
  description: string
  actif: boolean
}

/** جسم الطلب لإنشاء/تعديل نوع منشأة (POST/PUT /api/types-etablissement) */
export interface TypeEtablissementPayload {
  nom: string
  description: string
  actif: boolean
}
